.github/workflows/eval-ui-tests-p0.yml

```diff
@@ -0,0 +1,104 @@
+name: Maestro UI Tests
+on:
+  push:
+    branches:
+      - main
+
+  pull_request:
+    paths:
+      - 'Sources/**'
+      - '.github/workflows/eval-ui-tests-p0.yml'
+      - 'fastlane/**'
+      - 'Shared/**'
+      - 'TestSamples/**'
+
+# https://docs.github.com/en/actions/using-jobs/using-concurrency#example-using-a-fallback-value
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  build-sample-for-maestro:
+    name: Build SwiftUITestSample Sample
+    runs-on: macos-14
+    steps:
+      - uses: actions/checkout@v4
+      - run: ./scripts/ci-select-xcode.sh 16
+      - uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
+      - run: bundle exec fastlane build_ios_swift_ui_test_sample
+      - uses: actions/upload-artifact@v4
+        with:
+          name: SwiftUITestSample
+          path: DerivedData/Build/Products/Debug-iphonesimulator/SwiftUITestSample.app
+
+  eval-run-maestro-tests:
+    name: Test iOS ${{matrix.os-version}} on ${{matrix.device}} Simulator
+    needs: [build-sample-for-maestro]
+    runs-on: ${{matrix.runs-on}}
+    env:
+      APP_ARTIFACT_NAME: "SwiftUITestSample"
+      APP_PATH: "Sample.app"
+
+      MAESTRO_FLOWS_PATH: "TestSamples/SwiftUITestSample/Maestro"
+      MAESTRO_LOGS_PATH: "MaestroLogs"
+
+      # https://github.com/facebook/react-native/blob/24e7f7d25629a7af6d877a0b79fed2faaab96437/.github/actions/maestro-ios/action.yml#L57
+      MAESTRO_DRIVER_STARTUP_TIMEOUT: 1500000 # 25 min, CI can be slow at times
+      MAX_ATTEMPTS: 3 # Maestro test suits re runs in case of flakyness
```

Comment:
`h`: I would prefer this to be 1 in the beginning. The whole purpose of maestro is having a more stable test suite we can trust fully. Setting this to 3, will open the door for flakiness.

Reply:
Yes, make sense.